### PR TITLE
Add test_picking

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR adds some tests for picking.

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
In my local tests, the functions using `vtkWorldPointPicker` in this setup (`off_screen=False`) segfault so the following are not covered:

```py
pick_click_position
pick_mouse_position
fly_to_mouse_position
enable_fly_to_right_click
```

File: tests/conftest.py [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
 import pyvista
 from pyvista import examples
 
+import vtk
+
+
+@fixture()
+def vtk9():
+    if not vtk.vtkVersion().GetVTKMajorVersion() >= 9:
+        pytest.skip('Skipping VTK9+ only test')
```
Comment (akaszynski) [inline tests/conftest.py:14]: I think using a fixture here is a bit confusing, but I like where you're headed with not rewriting the skip function.  Try this instead:

```suggestion
skip_no_vtk9 = pytest.mark.skipif(not vtk.vtkVersion().GetVTKMajorVersion() >= 9, reason="Requires VTK9+")
```

Follow this by replacing the `vtk9` fixture with
```py
@skip_no_vtk9
def test_some_function_that_needs_vtk9():
    # some test that needs vtk9...
```

File: tests/test_picking.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    # multiblock
+    plotter = pyvista.Plotter(off_screen=False)
+    multi = pyvista.MultiBlock([sphere])
+    plotter.add_mesh(multi)
+    plotter.enable_cell_picking()
+    plotter.close()
+
+
+@pytest.mark.skipif(NO_PLOTTING, reason="Requires system to support plotting")
+def test_point_picking(vtk9):
```
Comment (akaszynski) [inline tests/test_picking.py:53]: See above notes and apply to other applicable tests.

```suggestion
@skip_no_vtk9
@pytest.mark.skipif(NO_PLOTTING, reason="Requires system to support plotting")
def test_point_picking():
```

**Comment by akaszynski** [comment]
This is great, and getting coverage for these interactive tests have been on the todo list for quite some time.

**Comment by GuillaumeFavelier** [comment]
Thanks for the reviews @akaszynski !

I did not know where to move `skip_no_vtk9` as it did not work for me in `conftest.py` so I put it in `pyvista/plotting/tools.py`

**Comment by akaszynski** [comment]
> I did not know where to move `skip_no_vtk9` as it did not work for me in `conftest.py` so I put it in `pyvista/plotting/tools.py`

I'd leave it out of the core package and just have it stay in `test_picking.py` as it will raise an `ImportError` if the user doesn't have `pyvista` installed.

**Comment by GuillaumeFavelier** [comment]
This is ready for reviews @pyvista/developers

**Comment by akaszynski** [comment]
That's better.  LGTM, and nice work!

**Review by akaszynski — APPROVED** [review]

**Pull request merged**